nomos-core/chain-defs/src/sdp/mod.rs

```diff
@@ -350,18 +354,89 @@ fn parse_length_prefixed_bytes(input: &[u8]) -> IResult<&[u8], Vec<u8>> {
     Ok((input, data.to_vec()))
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct BlendActivityProof {
+    pub session: SessionNumber,
+    #[serde(with = "serde_big_array::BigArray")]
+    pub proof_of_quota: [u8; PROOF_OF_QUOTA_SIZE],
+    pub proof_of_selection: [u8; PROOF_OF_SELECTION_SIZE],
+}
+
+const BLEND_ACTIVE_METADATA_VERSION_BYTE: u8 = 0x01;
+
+impl BlendActivityProof {
+    #[must_use]
+    pub fn to_metadata_bytes(&self) -> Vec<u8> {
+        let total_size = 1 // version byte
+            + size_of::<SessionNumber>()
+            + self.proof_of_quota.len()
+            + self.proof_of_selection.len();
+
+        let mut bytes = Vec::with_capacity(total_size);
+        bytes.push(BLEND_ACTIVE_METADATA_VERSION_BYTE);
+        bytes.extend(&self.session.to_le_bytes());
+        bytes.extend(&self.proof_of_quota);
+        bytes.extend(&self.proof_of_selection);
+        bytes
+    }
+
+    /// Parse metadata bytes using nom combinators
+    pub fn from_metadata_bytes(bytes: &[u8]) -> Result<Self, Box<dyn std::error::Error>> {
+        Ok(parse_blend_activity_proof(bytes)
+            .map_err(|e| format!("Failed to parse metadata: {e}"))?
+            .1)
+    }
+}
+
+fn parse_blend_activity_proof(input: &[u8]) -> IResult<&[u8], BlendActivityProof> {
+    let (input, version) = nom_u8(input)?;
+    if version != BLEND_ACTIVE_METADATA_VERSION_BYTE {
+        return Err(nom::Err::Error(nom::error::Error::new(
+            input,
+            nom::error::ErrorKind::Verify,
+        )));
+    }
+    let (input, session) = parse_session_number(input)?;
```

Comment:
Why noy storing the actual types instead of byte arrays in `BlendActivityProof`? Here all we know is that we can parse byte arrays with a given size. We don't know whether what's inside is actually a valid proof. Why not take the extra step of trying to convert the parsed bytes into actual `PoQ` and `PoSel`?

Reply:
Yeah. I thought about it. But, this `BlendActivityProof` is defined in the `nomos-core`. Whereas, `ProofOfQuota` and `ProofOfSelection` are defined in the `nomos-blend-message`. I thought about moving them into the `nomos-core` to remove duplication and byte arrays, but also didn’t want to move all the associated methods.

Reply:
I think I can move minimum components from `nomos-blend-message` to `nomos-core` to minimize duplication. To not make this PR too larger for other reviewers, I will do it as a separate PR.